sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1115,8 +1123,31 @@ object SQLContext {
             })
         }
     }
-    def createConverter(cls: Class[_], dataType: DataType): Any => Any = dataType match {
-      case struct: StructType => createStructConverter(cls, struct.map(_.dataType))
+    def createConverter(t: Type, dataType: DataType): Any => Any = (t, dataType) match {
+      case (cls: Class[_], struct: StructType) =>
+        createStructConverter(cls, struct.map(_.dataType))
+      case (arrayType: Class[_], array: ArrayType) =>
+        val converter = createConverter(arrayType.getComponentType, array.elementType)
+        value => new GenericArrayData(
+          (0 until JavaArray.getLength(value)).map(i =>
+            converter(JavaArray.get(value, i))).toArray)
+      case (_, array: ArrayType) =>
```

Comment:
Can you add few comments explaining why having two cases both for `ArrayType`?

Reply:
Sorry, I should have added a check for `cls.isArray` in the array case. That would make it clearer. I will also add a comment to each case with the actual type expected for that conversion.